### ISSUE TITLE
Feature request: validate local/relative links

### ISSUE THREAD

**Issue by agvxov** [igorshubovych/markdownlint-cli#657]
It would be useful to know whether there are any dead links in the document to files which should be present, but are not.

Example:
```md
See also [my other document](documents/my_other_document.md).

![demo](demo.gif)
```

Such files not being present is most definitely a fault of some kind.

The only possible issue I see is reliably determining what is supposed to be a file path and what is supposed to be an external link (without a protocol spec).

**Comment by DavidAnson** [comment]
Basically a duplicate of this new rule proposal: https://github.com/DavidAnson/markdownlint/issues/121
